### DISCUSSION TITLE
Question about your code for Node.js built-in modules loading

### DISCUSSION THREAD

**Discussion by nicolo-ribaudo** [denoland/deno#26835] [General]
Hey,

I was reading how Deno implements `require()` of built-in Node.js modules, and I noticed this file: https://github.com/denoland/deno/blob/main/ext/node/polyfills/01_require.js

The `punycode` module is deprecated, so it's defined as a getter with the deprecation warning:
https://github.com/denoland/deno/blob/7d326c269cfb8463d530ee5dd4bcabb7499e1ba7/ext/node/polyfills/01_require.js#L243-L251

However, that getter is triggered _immediately_ by https://github.com/denoland/deno/blob/7d326c269cfb8463d530ee5dd4bcabb7499e1ba7/ext/node/polyfills/01_require.js#L277, which immediately iterates through the object's entries triggering all the getters.

I'm asking this question because I don't actually see the warning unexpectedly, even if by reading the code it seems like the warning would be printed even if `punycode` is not required. What am I missing?

**Comment by bartlomieju** [comment]
Good catch, I bet this warning is emitted during V8 snapshot creation (ie. during the build process) and that's why it's never surfaced.
